pkg/cluster/EphemeralContainerService_test.go

```diff
@@ -0,0 +1,224 @@
+package cluster
+
+import (
+	"errors"
+	"github.com/devtron-labs/devtron/internal/util"
+	repository2 "github.com/devtron-labs/devtron/pkg/cluster/repository"
+	"github.com/devtron-labs/devtron/pkg/cluster/repository/mocks"
+	"github.com/go-pg/pg"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"testing"
+)
+
+func createSampleRequest() EphemeralContainerRequest {
+	return EphemeralContainerRequest{
+		BasicData: &EphemeralContainerBasicData{
+			ContainerName:       "container-1",
+			TargetContainerName: "target-container-1",
+			Image:               "image-1",
+		},
+		AdvancedData: &EphemeralContainerAdvancedData{
+			Manifest: "manifest-1",
+		},
+		Namespace: "namespace-1",
+		ClusterId: 1,
+		PodName:   "pod-1",
+		UserId:    123,
+	}
+}
+
+func TestForEphemeralContainers(t *testing.T) {
+
+	const (
+		namespace  = "namespace-1"
+		pod        = "pod-1"
+		containers = "container-1"
+	)
+
+	t.Run("TestSaveEphemeralContainer_Success", func(t *testing.T) {
+		repository := mocks.NewEphemeralContainersRepository(t)
+
+		repository.On("FindContainerByName", 1, namespace, pod, containers).Return(nil, nil)
+		repository.On("StartTx").Return(&pg.Tx{}, nil)
+		repository.On("RollbackTx", mock.AnythingOfType("*pg.Tx")).Return(nil) // Add this expectation
+		repository.On("SaveData", mock.AnythingOfType("*pg.Tx"), mock.AnythingOfType("*repository.EphemeralContainerBean")).Return(nil)
+		repository.On("SaveAction", mock.AnythingOfType("*pg.Tx"), mock.AnythingOfType("*repository.EphemeralContainerAction")).Return(nil)
+		repository.On("CommitTx", mock.AnythingOfType("*pg.Tx")).Return(nil)
+		logger, _ := util.NewSugardLogger()
+		service := NewEphemeralContainerServiceImpl(repository, logger)
+
+		request := createSampleRequest()
+
+		err := service.AuditEphemeralContainerAction(request, repository2.ActionCreate)
+
+		assert.NoError(t, err)
```

Comment:
Data Assertion should be there

Reply:
data assertion cannot be done here as this function only retutrns an error